### PR TITLE
Category as first argument for `AttributeCategoryForCAP` and `ActionsForCAP`

### PR DESCRIPTION
**Pull request by TKuh**
Part of #973.

### CODE REVIEW

File: AttributeCategoryForCAP/gap/AttributeCategory.gi [hunk 1]
```diff
@@ -233,9 +233,9 @@ InstallGlobalFunction( CAP_INTERNAL_INSTALL_ADDS_FOR_CATEGORY_WITH_ATTRIBUTES,
         
         output_source_getter := CAP_INTERNAL_METHOD_NAME_RECORD.(operation_name).output_source_getter;
         output_range_getter := CAP_INTERNAL_METHOD_NAME_RECORD.(operation_name).output_range_getter;
-        
+       
         return
-          function( arg )
+        function( cat, arg... )
```
Comment (zickgraf) [inline AttributeCategoryForCAP/gap/AttributeCategory.gi:238]: I think the old whitespace was correct.

File: AttributeCategoryForCAP/examples/example.g [hunk 2]
```diff
@@ -82,44 +82,44 @@ structure_record.morphism_type := TheTypeOfMorphismsOfObjectsWithEndomorphism;;
 
 ## This is the mathematical core: provide the functions that enhance objects with attributes
 structure_record.ZeroObject :=
-  function( underlying_zero_object )
+  function( cat, underlying_zero_object )
```
Comment (zickgraf) [inline AttributeCategoryForCAP/examples/example.g:85]: This is a backwards incompatible change, right? Does passing the category have a concrete advantage?

I agree that passing the category would be the "modern way" of doing things, but I would suggest not breaking the backwards compatibility without a concrete advantage. (Additionally, it looks like `cat` is the AttributeCategory but maybe the underlying category would make more sense.)

**Review by zickgraf — CHANGES_REQUESTED** [review]

File: AttributeCategoryForCAP/gap/AttributeCategory.gi (hunk 1 above)
Comment (TKuh) [reply AttributeCategoryForCAP/gap/AttributeCategory.gi]: Sorry, I missed that.

File: AttributeCategoryForCAP/examples/example.g (hunk 2 above)
Comment (zickgraf) [reply AttributeCategoryForCAP/examples/example.g]: My comment was meant to apply to all entries of `structure_record`. What is special about `ZeroObject` and `DualOnObjects` that you only reverted those two?

File: AttributeCategoryForCAP/examples/example.g [hunk 3]
```diff
@@ -119,7 +119,7 @@ structure_record.DualOnObjects :=
 ## In this particular example, this function would be not necessary since it can be 
 ## derived from the rigidity of the category
 structure_record.InternalHomOnObjects :=
-  function( object1, object2, underlying_internal_hom )
+  function( cat, object1, object2, underlying_internal_hom )
```
Comment (zickgraf) [inline AttributeCategoryForCAP/examples/example.g:122]: There is still a `cat` here, is this deliberate?

Comment (TKuh) [reply AttributeCategoryForCAP/examples/example.g]: Argh... no, it was probably too close to the `AddIsEqual...` below, so I overlooked it.
Thank you for spotting it.

**Review by zickgraf — APPROVED** [review]
Thanks!

**Pull request merged**